src/users/userSelectors.js

```diff
@@ -169,3 +179,122 @@ const getActiveUsersById: Selector<Map<UserId, UserOrBot>> = createSelector(
 // the `User` type definition.
 export const getUserIsActive = (state: PerAccountState, userId: UserId): boolean =>
   !!getActiveUsersById(state).get(userId);
+
+/** A user's value for a custom profile field, put into meaningful form. */
+export type CustomProfileFieldValue =
+  | { +displayType: 'text', +text: string }
+  | { +displayType: 'link', +text: string, +url: void | URL }
+  | { +displayType: 'users', +userIds: $ReadOnlyArray<UserId> };
+
+function interpretCustomProfileField(
+  realmDefaultExternalAccounts: $ElementType<RealmState, 'defaultExternalAccounts'>,
```

Comment:
> Could be `$PropertyType<RealmState, 'defaultExternalAccounts'>`, I think, since you're passing a string-literal type.

It could, but AFAICT `$PropertyType` didn't have any advantages over `$ElementType`, so it was better to just use a single name.

> or even `RealmState['defaultExternalAccounts']`!
> 
> Glad to be on newer Flow, since the RN v0.66 upgrade.

Oh, neat! Agreed.

Reply:
Ah, but it looks like our Prettier version doesn't support that syntax yet:
```
Running prettier...
prettier-eslint [ERROR]: prettier formatting failed due to a prettier error
prettier-eslint-cli [ERROR]: There was an error formatting "/home/greg/z/mobile/src/api/modelTypes.js": 
    SyntaxError: Unexpected token, expected "]" (312:24)
      310 |   // (We could be more specific here; but in the interest of reducing
      311 |   // differences between CrossRealmBot and User, just follow the latter.)
    > 312 |   +profile_data?: User['profile_data'],
```

Probably #5393 will fix that. I'll switch back for this PR.